### PR TITLE
MO-1472: Add health endpoint

### PR DESCRIPTION
**Pull request by ryantk**
Ticket: https://dsdmoj.atlassian.net/browse/MO-1472

- Added mechanism to define and carry out health checks
- Updated `/health` endpoint to output health check results

Deployed to test 2: https://test2.moic.service.justice.gov.uk/health

### CODE REVIEW

File: lib/health/status_check.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+module Health
+  module StatusCheck
+    def up?(get_response)
+      with_timeout { with_retries(2) { perform(get_response) } }
+    end
+
+  private
+
+    def with_timeout(&block)
+      Timeout.timeout(2, &block)
```
Comment (doctorpod) [inline lib/health/status_check.rb:10]: Never a bad idea to make naked values e.g. 2 constants e.g. TIMEOUT = 2, RETRIES = 2

File: spec/requests/info_spec.rb [hunk 2]
```diff
@@ -2,14 +2,13 @@
 
 describe "Info endpoinds" do
```
Comment (doctorpod) [inline spec/requests/info_spec.rb:3]: Not you, I know, but should this be 'endpoints'?

**Review by doctorpod — APPROVED** [review]
Couple of comments

File: lib/health/status_check.rb (hunk 1 above)
Comment (ryantk) [reply lib/health/status_check.rb]: @doctorpod I have switched the implementation from a module to a class with the timeouts/retries taken as parameters, this way is a bit cleaner I feel.

**Review by doctorpod — DISMISSED** [review]

File: config/initializers/health.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
+require 'health'
+
+config = Rails.configuration
+config.health_checks = Health.new(timeout_in_seconds_per_check: 2, num_retries_per_check: 2)
+  .add_check(
+    name: 'db',
+    get_response: -> { ActiveRecord::Base.connection.active? },
+    check_response: ->(response) { response == true })
+  .add_check(
+    name: 'priosnApi',
```
Comment (andrewrlee) [inline config/initializers/health.rb:10]: little typo

**Review by doctorpod — APPROVED** [review]

**Pull request merged**